### PR TITLE
Do not reset framework checkout before running customer tests

### PR DESCRIPTION
**Pull request by whesse**
The migration of customer tests to sharded tests adds a step that checks out the current tip-of-tree of the framework repo, removing local changes. This does not work with monorepo testing, which modifies engine.version, and does not work with local testing of a branch.

The sharded tests should already be running with the correct checkout of the framework repo. If the REVISION environment variable is set, the framework checkout will still be reset to check out that revision.

These commands were migrated from the existing shell script to the sharded tester in
https://github.com/flutter/flutter/pull/138659

Bug: https://github.com/dart-lang/sdk/issues/51042

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: dev/bots/test.dart [hunk 1]
```diff
@@ -1563,14 +1563,6 @@ Future<void> _runCustomerTesting() async {
     ],
     workingDirectory: flutterRoot,
   );
-  await runCommand(
```
Comment (godofredoc) [inline dev/bots/test.dart:1566]: Without this command the branch will be in detached mode, would the next steps succeed in detached mode?

Comment (whesse) [reply dev/bots/test.dart:1566]: The ci.sh command does work with the framework checkout in detached mode.

The command does require there to be a local branch called "master", so I have created that branch with the command

"git branch master origin/master"

Could we use origin/main instead?

File: dev/bots/test.dart [hunk 2]
```diff
@@ -1566,8 +1566,10 @@ Future<void> _runCustomerTesting() async {
   await runCommand(
     'git',
     <String>[
-      'checkout',
+      'branch',
+      '-f',
```
Comment (whesse) [inline dev/bots/test.dart:1570]: The -f force flag allows the command to succeed if the branch already exists.

File: dev/bots/test.dart (hunk 1 above)
Comment (godofredoc) [reply dev/bots/test.dart:1566]: in flutter/flutter main is a mirror of master. Using main may cause some race conditions.

**Review by godofredoc — APPROVED** [review]

**Pull request merged**